src/saml2/client_base.py

```diff
@@ -171,19 +171,17 @@ def __init__(self, config=None, identity_cache=None, state_cache=None,
             "authn_requests_signed": False,
             "want_assertions_signed": False,
             "want_response_signed": True,
-            "want_assertions_or_response_signed" : False
+            "want_assertions_or_response_signed": False,
         }
-
         for attr, val_default in attribute_defaults.items():
             val_config = self.config.getattr(attr, "sp")
-            if val_config is None:
-                val = val_default
-            else:
-                val = val_config
-
+            val = (
```

Comment:
This is not correct; `self.config.getattr(attr, "sp")` may return `False` which will end up giving `val_config` the `None` value. We must explicitly check for `None`.

Reply:
If fact "or None" means that if it's false It would rely to None. It's not tested but in my mind. It should be this way

Reply:
No. `None` means that the value is _unset_, not that it is `False`. In this case we must check the input from the user (the configuration) and explicitly check if the user has set a value.

If `val_default` is set to `True` and you set the configuration to `False` this will end up with `True`.

Reply:
I'm not on the pysaml2 behaviour, that's as It Is, no problem. I'm telling you that "False or None" would set None, at last, so that OR statement would work without all those if

Reply:
Yes, we're saying the same thing. But, what we want is to get `False`, not `None`. We want to know that the user (through the configuration) set a value. If we get `None`, then this is considered as "the user did not set a value" and thus we use the default. But, if the user has set a value (`False` in this example) then we want to use `False` - not the default value.